### ISSUE TITLE
Document on flow_from_* might need clarification

### ISSUE THREAD

**Issue by don-tpanic** [keras-team/keras-preprocessing#211]
I wonder if this is a known behaviour. So the document for `flow_from_directory` says "classes: Optional list of class subdirectories (e.g. ['dogs', 'cats']). Default: None. If not provided, the list of classes **will be automatically inferred from the subdirectory names/structure under directory, where each subdirectory will be treated as a different class (and the order of the classes, which will map to the label indices, will be alphanumeric**). The dictionary containing the mapping from class names to class indices can be obtained via the attribute class_indices." 

I think it is not entirely true, as the returned order of classes aren't actually sorted alphanumerically but in the order of the `classes`

Here is an example, I have three image folders called a, b, and c in `my_image_dir`
```
from keras.preprocessing.image import ImageDataGenerator

dir = 'my_image_dir'
sub_dirs = ['b', 'a']
datagen = ImageDataGenerator().flow_from_directory(
                                            directory=dir,
                                            batch_size=1,
                                            shuffle=False,
                                            class_mode="sparse",
                                            classes=sub_dirs,
                                            target_size=(224, 224)
                                            )
>>> datagen.class_indices = {'a': 1, 'b': 0}
```

Have I misunderstood something? Thanks.

**Comment by Dref360** [comment]
Not sure I understand your question. The order will be sorted if you do not provide the `classes` argument. In your example, you do provide this argument.

**Comment by don-tpanic** [comment]
> Not sure I understand your question. The order will be sorted if you do not provide the `classes` argument. In your example, you do provide this argument.

Yes was rather silly to ask now I realise. Thanks.
